test/hotspot/jtreg/compiler/vectorapi/VectorStoreMaskIdentityTest.java

```diff
@@ -0,0 +1,296 @@
+        testOneCastKernel(ShortVector.SPECIES_128, ByteVector.SPECIES_64);
+        verifyResult(ShortVector.SPECIES_128.length());
+
+        testTwoCastsKernel(ShortVector.SPECIES_64, IntVector.SPECIES_128, ShortVector.SPECIES_64);
+        verifyResult(ShortVector.SPECIES_64.length());
+
+        testThreeCastsKernel(ShortVector.SPECIES_128, ByteVector.SPECIES_64, ShortVector.SPECIES_128, ByteVector.SPECIES_64);
+        verifyResult(ShortVector.SPECIES_128.length());
+    }
+
+    @Test
+    @IR(counts = { IRNode.VECTOR_LOAD_MASK, "= 0",
+                   IRNode.VECTOR_STORE_MASK, "= 0",
+                   IRNode.VECTOR_MASK_CAST, "= 0" },
+        applyIfCPUFeatureOr = { "asimd", "true", "avx2", "true" },
+        applyIf = { "MaxVectorSize", "> 16" })
+    public static void testVectorMaskStoreIdentityShort256() {
+        testOneCastKernel(ShortVector.SPECIES_128, IntVector.SPECIES_256);
+        verifyResult(ShortVector.SPECIES_128.length());
+
+        testTwoCastsKernel(ShortVector.SPECIES_64, IntVector.SPECIES_128, LongVector.SPECIES_256);
+        verifyResult(ShortVector.SPECIES_64.length());
+
+        testThreeCastsKernel(ShortVector.SPECIES_128, ByteVector.SPECIES_64, FloatVector.SPECIES_256, IntVector.SPECIES_256);
+        verifyResult(ShortVector.SPECIES_128.length());
+    }
+
+    @Test
+    @IR(counts = { IRNode.VECTOR_LOAD_MASK, "= 0",
+                   IRNode.VECTOR_STORE_MASK, "= 0",
+                   IRNode.VECTOR_MASK_CAST, "= 0" },
+        applyIfCPUFeatureOr = { "asimd", "true", "avx", "true" },
+        applyIf = { "MaxVectorSize", ">= 16" })
+    public static void testVectorMaskStoreIdentityInt() {
+        testOneCastKernel(IntVector.SPECIES_MAX, FloatVector.SPECIES_MAX);
+        verifyResult(IntVector.SPECIES_MAX.length());
+
+        testTwoCastsKernel(IntVector.SPECIES_128, ShortVector.SPECIES_64, FloatVector.SPECIES_128);
+        verifyResult(IntVector.SPECIES_128.length());
+
+        testThreeCastsKernel(IntVector.SPECIES_128, ShortVector.SPECIES_64, FloatVector.SPECIES_128, ShortVector.SPECIES_64);
+        verifyResult(IntVector.SPECIES_128.length());
+    }
+
+    @Test
+    @IR(counts = { IRNode.VECTOR_LOAD_MASK, "= 0",
+                   IRNode.VECTOR_STORE_MASK, "= 0",
+                   IRNode.VECTOR_MASK_CAST, "= 0" },
+        applyIfCPUFeatureOr = { "asimd", "true", "avx2", "true" },
+        applyIf = { "MaxVectorSize", "> 16" })
+    public static void testVectorMaskStoreIdentityInt256() {
+        testOneCastKernel(IntVector.SPECIES_MAX, FloatVector.SPECIES_MAX);
+        verifyResult(IntVector.SPECIES_MAX.length());
+
+        testTwoCastsKernel(IntVector.SPECIES_128, ShortVector.SPECIES_64, LongVector.SPECIES_256);
+        verifyResult(IntVector.SPECIES_128.length());
+
+        testThreeCastsKernel(IntVector.SPECIES_128, ShortVector.SPECIES_64, FloatVector.SPECIES_128, LongVector.SPECIES_256);
+        verifyResult(IntVector.SPECIES_128.length());
+    }
```

Comment:
A nit, you can define final static species like S128, S64 pointing to fully qualified species, it will reduce the verbosity.